src/typechecker.erl

```diff
@@ -136,14 +138,20 @@ compat(T1, T2, A, TEnv) ->
             compat_ty(Ty1, Ty2, sets:add_element({Ty1, Ty2}, A), TEnv)
     end.
 
-%% any() is used as the unknown type in the gradual type system
+%% any() and term() are used as the unknown type in the gradual type system
```

Comment:
If we normalize term to any, we don't need to handle term here and in some other places.

Reply:
That's a good point. We should start moving to a point where the internal representation we have for types is different from the type representation used in the absform. Normalizing away `term()` in favor of `any()` is one such thing, and we'll also want to use a more efficient representation of unions to solve the efficiency problems with have with it.
I'll keep the code as it is for now, and we can clean it up as part of being more stringent about the separation between our internal representation and the absform representation.